Rater.Business/Services/SpaceService.cs

```diff
@@ -12,25 +16,48 @@ namespace Rater.Business.Services
     public class SpaceService : ISpaceService
     {
 
-        private readonly ISpaceRepository _repo;
-        public SpaceService(ISpaceRepository repo)
+        private readonly ISpaceRepository _spaceRepo;
+        private readonly IUserRepository _userRepo;
+        private readonly IMapper _mapper;
+        public SpaceService(ISpaceRepository spaceRepo,IUserRepository userRepo,IMapper mapper)
         {
-            _repo = repo;
+            _spaceRepo = spaceRepo;
+            _userRepo = userRepo;
+            _mapper = mapper;
         }
 
 
 
         public async Task<List<SpaceResponseDto>> GetAllSpaces()
         {
-            var value = await _repo.GetAllSpaces();
+            var value = await _spaceRepo.GetAllSpaces();
             return value;
         }
 
-        public async Task<SpaceResponseDto> CreateSpace(SpaceRequestDto request)
+        public async Task<SpaceResponseDto> AddSpace(GrandSpaceRequestDto request)
         {
-            var value = await _repo.CreateSpace(request);
-            return value;
+            UserRequestDto userRequest = new UserRequestDto();
+            userRequest.NickName = request.creatorNickname;
+            var justCreatedUser = await _userRepo.AddUser(userRequest);
+
+
+            var space = _mapper.Map<Space>(request);
+            space.CreatorId = await _userRepo.GetCreatorId(justCreatedUser.CreatedAt);
+
+            foreach (var metrics in space.Metrics)
```

Comment:
Consider using [LINQ](https://blog.lahteenmaki.net/java-streams-vs-c-linq-vs-java6.html) instead of a for-loop. 

>Why should LINQ be faster? It also uses loops internally.
>
>Most of the times, LINQ will be a bit slower because it introduces overhead. Do not use LINQ if you care much about performance. Use LINQ because you want shorter better readable and maintainable code.
> https://stackoverflow.com/questions/3156059/is-a-linq-statement-faster-than-a-foreach-loop